### ISSUE TITLE
Potential rugplot speedup using ax.h/vlines instead of ax.axh/vline

### ISSUE THREAD

**Issue by mroeschke** [mwaskom/seaborn#1368]
I noticed that `sns.distplot(..., rug=True)` is a can be a little slow when passing in a larger sequence of data.

```
In [1]: import seaborn as sns

In [2]: data = list(range(1000))

In [3]: %timeit sns.distplot(data, rug=True)
1 loop, best of 3: 1.58 s per loop
```

And I think this is because `rugplot` is plotting each line is a for loop here:
https://github.com/mwaskom/seaborn/blob/88c49ef6c14ca69def8195f03d467915b3b0597c/seaborn/distributions.py#L692-L695

In [this branch](https://github.com/mroeschke/seaborn/commit/9f146807e0173f889cfc956bcde47ab521f5ae30) where I replace `ax.axhline` with `ax.hlines`, I got a nice speedup:

```
In [1]: import seaborn as sns

In [2]: data = list(range(1000))

In [3]: %timeit sns.distplot(data, rug=True)
The slowest run took 4.16 times longer than the fastest. This could mean that an intermediate result is being cached.
10 loops, best of 3: 21.8 ms per loop
```

However the ymax/min parameter in `ax.hlines` refers to absolute height while `ax.axhline` is more of a scale. I was curious if you knew how to convert between the two (possibly a good question for the matplotlib folks too), as this would be a nice little speedup for the `rugplot` function.

Thanks for all your work on this package!

**Comment by mroeschke** [comment]
Thanks for the insight! For my use case, I occasionally use the rugplot to exactly visualize where the outliers lie in a larger dataset when looking at the distribution.

Definitely a low priority request but would be much appreciated in a future version of seaborn. Thanks!

**Comment by mwaskom** [comment]
Closed with #1415
